### PR TITLE
Ba/feature/per collection metadata fields

### PR DESCRIPTION
**Pull request by bradleyandrick**
**Related Issue(s):**

- NA

**Proposed Changes:**

1. Change Popup Result component to only render title (id of stac item) as default field
2. Add setting in `config.json` to allow for user defined properties to specify if additional metadata fields should be set in Popup Result

> NOTE: This PR improves upon configuration but still couples the metadata fields to some known properties needed for demo UI projects. There is a followup ticket created by project team to refactor this in more detail and introduce things like field name aliases and better handle presentation of edge cases with varying data types and human readable formatting needs.  

### To Test:

**automated tests pass** 🍏  `npm run test`

**Test config set**
- add `POPUP_DISPLAY_FIELDS` setting to local `config.json` to match needs and formatting found in `config.example.json`
- load app
- click search and zoom until scenes are showing on map
- click footprints
- confirm popup results show
- scroll in list and confirm popup results display as
  - strings as strings
  - numbers as strings
  - bools as text strings
  - array values as comma separated strings
  - `eo:cloud_cover` is handled uniquely with a truncate to two decimal places and a concatenated `%`
- change between collections and confirm metadata gets updated to match config settings

**Test config partially set**
- stop app
- remove one of the collection values from `POPUP_DISPLAY_FIELDS` but leave the others
- load app
- select collection from search dropdown in top left of search/filters bar that you removed from the `POPUP_DISPLAY_FIELDS` config settings
- click search and zoom until scenes are showing on map
- click footprints
- confirm popup results show
- confirm only title field is showing for this collection
- change to another collection that still exists in `POPUP_DISPLAY_FIELDS` config settings
- click search and zoom until scenes are showing on map
- click footprints
- confirm popup results show
- confirm metadata still gets updated to match config settings

**Test no config**
- stop app
- remove `POPUP_DISPLAY_FIELDS` setting from local `config.json` 
- load app
- click search and zoom until scenes are showing on map
- click footprints
- confirm popup results show
- confirm only title field is showing for this collection
- change to another collection
- confirm only title field is still only field showing

**PR Checklist:**

- [X] I have added my changes to the [CHANGELOG](https://github.com/Element84/filmdrop-ui/blob/main/CHANGELOG.md) **or** a CHANGELOG entry is not required.

### CODE REVIEW

File: public/config/config.example.json [hunk 1]
```diff
@@ -101,5 +101,19 @@
...
+    "landsat-c2-l2": ["datetime", "platform", "instruments", "eo:cloud_cover"],
+    "naip": ["datetime", "naip:state", "naip:year", "gsd"],
+    "cop-dem-glo-30": ["datetime"],
+    "cop-dem-glo-90": ["datetime"],
+    "sentinel-1-grd": [
+      "datetime",
+      "platform",
+      "sar:instrument_mode",
+      "sar:polarizations"
+    ]
```
Comment (stephenkilbourn) [inline public/config/config.example.json:117]: not applicable to the PR, but it'd be nice if there was something documenting the available fields.  a little zod or typescript like to make sure there aren't typos.

**Review by stephenkilbourn — APPROVED** [review]

File: public/config/config.example.json (hunk 1 above)
Comment (bradleyandrick) [reply public/config/config.example.json:117]: good note. one of the challenges here is that this configuration could handle any fields from any stac apis. There might be a decent way to document some of that but probably need a brainstorming session to figure out a useable approach.

File: src/components/PopupResult/PopupResult.jsx [hunk 2]
```diff
@@ -80,40 +81,32 @@ const PopupResult = (props) => {
...
+                      className="popupResultDetailsRowKey"
+                      key={field + '2'}
+                    >
+                      {field.charAt(0).toUpperCase() + field.slice(1) + ':'}
+                    </span>
+                    <span
+                      className="popupResultDetailsRowValue"
+                      key={field + '3'}
+                    >
+                      {field === 'eo:cloud_cover'
```
Comment (philvarner) [inline src/components/PopupResult/PopupResult.jsx:99]: consider replacing this special case with additional config for `decimal_places` (which use a proper truncation function to clip the places rather than this */ 100 trick) and a `units` config for the `%`

**Review by philvarner — APPROVED** [review]
